frontend/src/utils/TaskUtils.ts

```diff
@@ -305,134 +267,40 @@ export const isRatedByCustomer = (
 */
 export const isUnrated = (
   user: RoadmapUser | Customer | UserInfo,
-  customers?: Customer[],
-) => (task: Task) => {
+  roadmap: Roadmap,
+): ((task: Task) => boolean) => {
   if (isCustomer(user))
-    return !!user.representatives?.find(
-      (rep) => !task.ratings.some(isRatedByCustomer(user, rep)),
-    );
+    return (task) =>
+      !!user.representatives?.some((rep) => !ratedByCustomer(user, rep)(task));
+
   if (isUserInfo(user)) {
-    const type = getType(user.roles, task.roadmapId);
-    if (type === RoleType.Admin || type === RoleType.Business)
-      return !!user.representativeFor?.find(
-        (customer) => !task.ratings.some(isRatedByCustomer(customer, user)),
-      );
-    return !task.ratings.find(isRatedByUser(user));
+    return getType(user.roles, roadmap.id) === RoleType.Admin
+      ? hasMissingRatings(roadmap)
+      : awaitsUserRatings(user, roadmap);
   }
+
   if (user.type === RoleType.Admin || user.type === RoleType.Business) {
-    const represented = customers?.filter((customer) =>
-      customer.representatives?.find((rep) => rep.id === user.id),
-    );
-    return !!represented?.find(
-      (customer) => !task.ratings.some(isRatedByCustomer(customer, user)),
+    const represented = roadmap.customers?.filter((customer) =>
```

Comment:
Maybe clients or customers would suit better here too (at least I thought 'wait who are represented' for a second)
```suggestion
    const clients = roadmap.customers?.filter((customer) =>
```